controllers/dspipeline_params.go

```diff
@@ -632,6 +634,14 @@ func (p *DSPAParams) ExtractParams(ctx context.Context, dsp *dspa.DataSciencePip
 			}
 		}
 
+		if p.APIServer.CABundleFileMountPath != "" {
+			p.CustomCABundleRootMountPath = p.APIServer.CABundleFileMountPath
+		}
+		if p.APIServer.CABundle != nil && p.APIServer.CABundle.ConfigMapKey != "" {
+			dspTrustedCAConfigMapKey = p.APIServer.CABundle.ConfigMapKey
+		}
+		p.PiplinesCABundleMountPath = fmt.Sprintf("%s/%s", p.CustomCABundleRootMountPath, dspTrustedCAConfigMapKey)
```

Comment:
I think you have confused the user provided configmap with the one we create. 

User provides a configmap [here](https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/api/v1alpha1/dspipeline_types.go#L77). We use this configmap (and another configmap "odh-trusted-ca-bundle" if provided), and combine them to create a new configmap [here](https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/controllers/dspipeline_params.go#L646-L671). It's this 2nd configmap that we create, which we want to allow the user to _also_ control but this is unrelated to the configmap key provided via `p.APIServer.CABundle.ConfigMapKey`.  

`p.APIServer.CABundle` is the user saying "please create a new cabundle, but include my cabundle found at this specified location". 

`.spec.apiServer.CABundleFileName` is the user saying, "when you create this new cabundle, please put it in this folder". 

We want a new field that says, "when you create this new cabundle, please put it in this folder, and use **this filename**". 

For this we need to introduce a new crd `.spec.apiServer.CABundleFileName` with a description: 

> CABundleFileName is the name given to the CABundle created in the path CaBundleFileMountPath (default dspa.crt)

Use this to populate `dspTrustedCAConfigMapKey` much like you do `p.CustomCABundleRootMountPath`

Reply:
be sure to also add this new field to: https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/config/samples/v2/dspa-all-fields/dspa_all_fields.yaml